ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/operations/OperationSignatureVerifier.java

```diff
@@ -107,4 +109,29 @@ public boolean verifyVoluntaryExitSignature(
     final Bytes signingRoot = miscHelpers.computeSigningRoot(exit, domain);
     return signatureVerifier.verify(maybePublicKey.get(), signingRoot, signedExit.getSignature());
   }
+
+  public boolean verifyBlsToExecutionChangeSignature(
+      final Fork fork,
+      final BeaconState state,
+      final SignedBlsToExecutionChange signedBlsToExecutionChange,
+      final BLSSignatureVerifier signatureVerifier) {
+    final BlsToExecutionChange addressChange = signedBlsToExecutionChange.getMessage();
+    final int validatorIndex = addressChange.getValidatorIndex().intValue();
+    if (state.getValidators().size() <= validatorIndex || validatorIndex < 0) {
+      return false;
+    }
+    if (beaconStateAccessors.validateWithdrawalCredentials(state, addressChange)) {
```

Comment:
`validateWithdrawalCredentials` is duplicating these validatorIndex checks. But in the signature verifier I don't think we should be calling validateWithdrawalCredentials at all. At which point I don't think we need to get the validator at all so no need to check the index here.